### PR TITLE
move selectedStory out of redux and into route

### PR DESCRIPTION
**Pull request by shc023**
- clicking card changes routes and opens modal
- closing modal changes route back to `projects/<id>`
- going directly to `projects/<id>?story=<storyId>` loads the board and opens the right modal

### CODE REVIEW

**Comment by mattvv** [comment]
Gonna test this a bit tomorrow Scott, don't merge yet

**Comment by mattvv** [comment]
@mikealche this is pretty wildy out of date, but wanted to get your thoughts on why shallow render is still causing all the cards to re-render.

File: pages/projects/[id].tsx [hunk 1]
```diff
@@ -45,32 +52,29 @@ const Project: NextPage = (): JSX.Element => {
...
-    setFilters({ ...filters, [name]: getFilterArray(filters[name], filter) });
-  };
+  const selectedStory = useSelector((state: State): Story => getStory(state, id, selectedStoryId));
+
+  const addFilter = useCallback(
+    (name: string, filter: Owner | Label | Iteration): void => {
+      if (name === 'iterations') {
+        //todo: we should change this check to a type check and uses classes so we don't have to do this.
+        // @ts-ignore: we know iteration can only be Iteration type
+        setFilters({ ...filters, iteration: filter });
```
Comment (diegofigs) [inline pages/projects/[id].tsx:70]: Think you can use type predicates or `[instanceof](https://www.typescriptlang.org/docs/handbook/advanced-types.html#instanceof-type-guards)` operation to type guard
```suggestion
        setFilters({ ...filters, iteration: (filter as Iteration) });
```

File: pages/projects/[id].tsx [hunk 2]
```diff
@@ -45,32 +52,29 @@ const Project: NextPage = (): JSX.Element => {
...
+        return;
+      }
+      // @ts-ignore: we know filter cannot be type Iteration
+      setFilters({ ...filters, [name]: getFilterArray(filters[name], filter) });
```
Comment (diegofigs) [inline pages/projects/[id].tsx:74]: Same here

**Review by diegofigs — APPROVED** [review]

**Review by mattvv — CHANGES_REQUESTED** [review]
just leaving this as request changes until we figure out the performance issue

**Pull request closed without merging**